model/integration_action.go

```diff
@@ -0,0 +1,260 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package model
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"encoding/asn1"
+	"encoding/base64"
+	"encoding/json"
+	"io"
+	"math/big"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+const (
+	POST_ACTION_TYPE_BUTTON                         = "button"
+	POST_ACTION_TYPE_SELECT                         = "select"
+	INTERACTIVE_DIALOG_TRIGGER_TIMEOUT_MILLISECONDS = 3000
+)
+
+type DoPostActionRequest struct {
+	SelectedOption string `json:"selected_option"`
+}
+
+type PostAction struct {
+	Id          string                 `json:"id"`
+	Name        string                 `json:"name"`
+	Type        string                 `json:"type"`
+	DataSource  string                 `json:"data_source"`
+	Options     []*PostActionOptions   `json:"options"`
```

Comment:
I assume this is for Slack compatibility. Otherwise a `map[string]string` would make more sense.

Reply:
It needs to be ordered so a map[string]string wouldn't work

Reply:
Why does it need to be ordered?

Reply:
Because it controls what order the options are shown on the client